tests/foreman/virtwho/test_cli_virtwho.py

```diff
@@ -127,17 +133,23 @@ def test_positive_deploy_configure_by_script(self):
         self.assertEquals(
             VirtWhoConfig.info({'id': vhd['id']})['general-information']['status'], 'OK')
         hosts = [
-            (hypervisor_name, 'product_id={}'.format(self.vdc_physical)),
-            (guest_name, 'type=STACK_DERIVED')]
+            (hypervisor_name, 'product_id={} and type=NORMAL'.format(self.vdc_physical)),
+            (guest_name, 'product_id={} and type=STACK_DERIVED'.format(self.vdc_physical))]
         for hostname, sku in hosts:
             host = Host.list({'search': hostname})[0]
-            vdc = Subscription.list({
+            subscriptions = Subscription.list({
                 'organization': DEFAULT_ORG,
                 'search': sku,
-            })[0]
+            })
+            vdc_id = subscriptions[0]['id']
+            if 'type=STACK_DERIVED' in sku:
+                for item in subscriptions:
+                    if hypervisor_name in item['type']:
+                        vdc_id = item['id']
+                        break
```

Comment:
This condition and loop is present four times, and almost identical in all cases. Would it be possible to use helper function instead?

Reply:
As above, api and cli are difference condition, the helper function will be limited.